backend/coins/btc/transactions/transactions_test.go

```diff
@@ -190,6 +191,39 @@ func (s *transactionsSuite) TestUpdateAddressHistorySingleTxReceive() {
 	require.Equal(s.T(), expectedHeight, transactions[0].Height)
 }
 
+// TestTxSerialization checks that Tx marshaling/unmarshaling work with both current and legacy
+// chainhash.Hash marshaling methods.
+// see https://github.com/btcsuite/btcd/pull/2025.
+func (s *transactionsSuite) TestTxSerialization() {
+	// create a new Tx
+	addresses, err := s.addressChain.EnsureAddresses()
+	require.NoError(s.T(), err)
+	address := addresses[0]
+	expectedAmount := btcutil.Amount(123)
+	tx1 := newTx(chainhash.HashH(nil), 0, address, expectedAmount)
+
+	// marshal & unmarshal transaction with current method.
+	tx1Serialized, err := json.Marshal(tx1)
+	require.NoError(s.T(), err)
+	s.log.Print("Tx marshaled: " + string(tx1Serialized))
+	var tx2 *wire.MsgTx
+	err = json.Unmarshal(tx1Serialized, &tx2)
+	require.NoError(s.T(), err)
+
+	// marshal prevout hash with legacy method.
+	hashBytes := [32]byte(tx1.TxIn[0].PreviousOutPoint.Hash.CloneBytes())
+	legacyMarshalledHash, err := json.Marshal(hashBytes)
+	require.NoError(s.T(), err)
+
+	// build a serialized tx string which is identical to the previous tx, except for the prevout hash and unmarshal it.
+	tx2serialized := "{\"Version\":1,\"TxIn\":[{\"PreviousOutPoint\":{\"Hash\":" + string(legacyMarshalledHash) + ",\"Index\":0},\"SignatureScript\":null,\"Witness\":null,\"Sequence\":4294967295}],\"TxOut\":[{\"Value\":123,\"PkScript\":\"dqkU+tJNI2oRHKW1g4q4XbLEvaq3w76IrA==\"}],\"LockTime\":0}"
```

Comment:
asked chatgpt to make it a multiline string literal

```Go
tx2serialized := `{
	"Version": 1,
	"TxIn": [{
		"PreviousOutPoint": {
			"Hash": ` + string(legacyMarshalledHash) + `,
			"Index": 0
		},
		"SignatureScript": null,
		"Witness": null,
		"Sequence": 4294967295
	}],
	"TxOut": [{
		"Value": 123,
		"PkScript": "dqkU+tJNI2oRHKW1g4q4XbLEvaq3w76IrA=="
	}],
	"LockTime": 0
}`
```

nicer to read? :smile:

Reply:
Nice! Thanks